Add 5 Detection Methods

MMJv3 opened the pull request:
Seems like the script as of now doesn't check these values for whats stored in them, which is important because the values are set to blatant vmware values.

HKEY_LOCAL_MACHINE\HARDWARE\DESCRIPTION\System\BIOS\BIOSVendor
HKEY_LOCAL_MACHINE\HARDWARE\DESCRIPTION\System\BIOS\BIOSVersion

I edited the script to add real data that would be found in a PC in place of the vmware one, I'm not sure if that might conflict with the 'BaseBoard' Values (The manufacturer, product, and version) and since everyone's PC is different.

Hoping you can take a look at this and see if its something that should be changed (and if random values would work in place for that registry key)

top is the VM obviously

![image](https://github.com/user-attachments/assets/6f98c055-e15b-4afd-973d-6de737266840)

MMJv3 committed:
Update VMwareCloak.ps1

MMJv3 committed:
Update VMwareCloak.ps1

MMJv3 commented:
ignore the extra commit... spelling error..

d4rksystem commented:
> Seems like the script as of now doesn't check these values for whats stored in them, which is important because the values are set to blatant vmware values.
> 
> HKEY_LOCAL_MACHINE\HARDWARE\DESCRIPTION\System\BIOS\BIOSVendor HKEY_LOCAL_MACHINE\HARDWARE\DESCRIPTION\System\BIOS\BIOSVersion
> 
> I edited the script to add real data that would be found in a PC in place of the vmware one, I'm not sure if that might conflict with the 'BaseBoard' Values (The manufacturer, product, and version) and since everyone's PC is different.
> 
> Hoping you can take a look at this and see if its something that should be changed (and if random values would work in place for that registry key)
> 
> top is the VM obviously
> 
> ![image](https://private-user-images.githubusercontent.com/36246145/400602688-6f98c055-e15b-4afd-973d-6de737266840.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY0NDkzMzYsIm5iZiI6MTczNjQ0OTAzNiwicGF0aCI6Ii8zNjI0NjE0NS80MDA2MDI2ODgtNmY5OGMwNTUtZTE1Yi00YWZkLTk3M2QtNmRlNzM3MjY2ODQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA5VDE4NTcxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc3MmNjYjY0NGY1MmMwZGQwN2ZmZTJhYzg3ZTAxYTMwZWRlYzVjZWY4ZWRhZmI3MWJlYjE3NTM3NjI3MDZiNjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7wAo29UHfpOal2tDOmFZ8giKwDxuaQuT6asS6

d4rksystem commented:
Awesome, thanks @MMJv3! I'll take a look this week/weekend and test. Once I complete testing, I'll merge and give you credit :)

MMJv3 committed:
Update VMwareCloak.ps1

MMJv3 commented:
latest commit

call me crazy but i did a deeper dive, found some more. will update the ones I think are actually useful to be added.

HKEY_LOCAL_MACHINE\HARDWARE\DEVICEMAP\Scsi\Scsi Port 3\Scsi Bus 1\Target Id 0\Logical Unit Id 0\Identifier

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\Class\{4d36e97d-e325-11ce-bfc1-08002be10318}\0133\DriverDesc

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\Class\{4d36e97d-e325-11ce-bfc1-08002be10318}\0133\InfSection

(not too sure about the InfSection above, added it since the value looked suspect)

**The ones below I have way less confidence will trigger anything, I don't know any malware or program that would do this deep of a dive just to make sure you arent on a VM. I added them for posterity's sake to the comment, but not the PR**

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\DeviceClasses{53f56307-b6bf-11d0-94f2-00a0c91efb8b}##?#SCSI#Disk&VenVMware&Prod_VMware_Virtual_S#5&1ec51bf7&0&000000#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}\Deviceinstance

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\DeviceClasses{53f56308-b6bf-11d0-94f2-00a0c91efb8b}##?#SCSI#CdRom&Ven_NECVMWar&Prod_VMware_SATA_CD01#5&260e6d66&0&010000#{53f56308-b6bf-11d0-94f2-00a0c91efb8b}\Deviceinstance

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\DeviceClasses{53f56308-b6bf-11d0-94f2-00a0c91efb8b}##?#SCSI#CdRom&Ven_NECVMWar&Prod_VMware_SATA_CD01#5&260e6d66&0&010000#{53f56308-b6bf-11d0-94f2-00a0c91efb8b}\Deviceinstance

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\DeviceClasses{7fccc86c-228a-40ad-8a58-f590af7bfdce}##?#SCSI#Disk&VenVMware&Prod_VMware_Virtual_S#5&1ec51bf7&0&000000#{7fccc86c-228a-40ad-8a58-f590af7bfdce}\Deviceinstance

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\Video{FC1F882D-CCA2-11EF-904F-806E6F6E6963}\0000\HardwareInformation.ChipType

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\Video{FC1F882D-CCA2-11EF-904F-806E6F6E6963}\0000\HardwareInformation.DacType

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Enum\PCI\VEN_15AD&DEV_0740&SUBSYS_074015AD&REV_10\3&61aaa01&0&3F\DeviceDesc

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Enum\SCSI\CdRom&Ven_NECVMWar&Prod_VMware_SATA_CD01\5&260e6d66&0&010000\FriendlyName

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Enum\SCSI\CdRom&Ven_NECVMWar&Prod_VMware_SATA_CD01\5&260e6d66&0&010000\HardwareID

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Enum\SCSI\Disk&Ven_VMware_&Prod_VMware_Virtual_S\5&1ec51bf7&0&000000\FriendlyName

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Enum\SCSI\Disk&Ven_VMware_&Prod_VMware_Virtual_S\5&1ec51bf7&0&000000\HardwareID

d4rksystem commented:
> latest commit
> 
> call me crazy but i did a deeper dive, found some more. will update the ones I think are actually useful to be added.
> 
> HKEY_LOCAL_MACHINE\HARDWARE\DEVICEMAP\Scsi\Scsi Port 3\Scsi Bus 1\Target Id 0\Logical Unit Id 0\Identifier
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\Class{4d36e97d-e325-11ce-bfc1-08002be10318}\0133\DriverDesc
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\Class{4d36e97d-e325-11ce-bfc1-08002be10318}\0133\InfSection
> 
> (not too sure about the InfSection above, added it since the value looked suspect)
> 
> **The ones below I have way less confidence will trigger anything, I don't know any malware or program that would do this deep of a dive just to make sure you arent on a VM. I added them for posterity's sake to the comment, but not the PR**
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\DeviceClasses{53f56307-b6bf-11d0-94f2-00a0c91efb8b}##?#SCSI#Disk&VenVMware&Prod_VMware_Virtual_S#5&1ec51bf7&0&000000#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}\Deviceinstance
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\DeviceClasses{53f56308-b6bf-11d0-94f2-00a0c91efb8b}##?#SCSI#CdRom&Ven_NECVMWar&Prod_VMware_SATA_CD01#5&260e6d66&0&010000#{53f56308-b6bf-11d0-94f2-00a0c91efb8b}\Deviceinstance
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\DeviceClasses{53f56308-b6bf-11d0-94f2-00a0c91efb8b}##?#SCSI#CdRom&Ven_NECVMWar&Prod_VMware_SATA_CD01#5&260e6d66&0&010000#{53f56308-b6bf-11d0-94f2-00a0c91efb8b}\Deviceinstance
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\DeviceClasses{7fccc86c-228a-40ad-8a58-f590af7bfdce}##?#SCSI#Disk&VenVMware&Prod_VMware_Virtual_S#5&1ec51bf7&0&000000#{7fccc86c-228a-40ad-8a58-f590af7bfdce}\Deviceinstance
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\Video{FC1F882D-CCA2-11EF-904F-806E6F6E6963}\0000\HardwareInformation.ChipType
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\Video{FC1F882D-CCA2-11EF-904F-806E6F6E6963}\0000\HardwareInformation.DacType
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Enum\PCI\VEN_15AD&DEV_0740&SUBSYS_074015AD&REV_10\3&61aaa01&0&3F\DeviceDesc
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Enum\SCSI\CdRom&Ven_NECVMWar&Prod_VMware_SATA_CD01\5&260e6d66&0&010000\FriendlyName
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Enum\SCSI\CdRom&Ven_NECVMWar&Prod_VMware_SATA_CD01\5&260e6d66&0&010000\HardwareID
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Enum\SCSI\Disk&Ven_VMware_&Prod_VMware_Virtual_S\5&1ec51bf7&0&000000\FriendlyName
> 
> HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Enum\SCSI\Disk&Ven_VMware_&Prod_VMware_Virtual_S\5&1ec51bf7&0&000000\HardwareID

Sorry for the delay on my part! Still going through your additions and testing :) One thing I've noticed is that a lot of the reg keys that have a path like "7fccc86c-228a-40ad-8a58-f590af7bfdce" are randomized by the OS. So these will be different on my VM. If my understanding is correct, these would not be good keys to include in the script. Please correct me if I'm wrong here.

One thing we could do is just do a simple search/replace for all reg values that have the string "VMware" or similar. I thought about doing this in the past and I tested this quite a bit. However, it broke some features of the VM so I decided against this. Also, malware is mainly looking for  hardcoded paths and not every reg value that has the string "VMware". 

Hope that makes sense! Open to your thoughts on this :)

Thanks a lot for the contribution - please give me some time to go through it all :)

d4rksystem commented:
I went through all your additions and they look great. Tested on my end and all looks good. I really appreciated your detective work in finding new VMware registry keys! :) 

A couple notes:

- You were right about the class ID's (such as "4d36e97d-e325-11ce-bfc1-08002be10318"). It seems these are not randomized and they also exist on my VM's.
- Regarding the 7 other registry keys, Would you be open to submitting a new PR with these (as soon as I merge this one)? I looked at the reg keys in my VM's and they match. What I would propose is we just rename the reg key itself, not the value. For example, rename the enture "HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\DeviceClasses{53f56307-b6bf-11d0-94f2-00a0c91efb8b}##?#SCSI#Disk&VenVMware&Prod_VMware_Virtual_S#5&1ec51bf7&0&000000#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}" key, not the "Deviceinstance" value. Since the key itself contains the word VMware :)
- You "HKEY_LOCAL_MACHINE\HARDWARE\DEVICEMAP\Scsi\Scsi Port 3\Scsi Bus 1\Target Id 0\Logical Unit Id 0\Identifier" key didn't match my VM's. I did some research on this and found that this is randomized by VMware. So the VM keys could be in "Scsi Port 0" or "Scsi Bus 3" for example. I kept your addition anyway, because other people's VM's may look the same as yours.
